confighelper/beaconbroadcaster.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2017-2018 Joel Allen Luellwitz and Andrew Klapp
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Provides the broadcasting component of a a filesystem-based IPC mechanism."""
+
+__all__ = ['BeaconBroadcasterBroadcastException',
+           'BeaconBroadcasterInitException',
+           'BeaconBroadcaster']
+
+import datetime
+import logging
+import os
+import stat
+import tmpfs
+
+GROUP_RW_MODE = stat.S_IXUSR | stat.S_IRUSR | stat.S_IWUSR | stat.S_IXGRP | stat.S_IRGRP
+GROUP_RO_MODE = stat.S_IXUSR | stat.S_IRUSR | stat.S_IRGRP
+RW_MODE = stat.S_IXUSR | stat.S_IRUSR | stat.S_IWUSR
+
+BEACON_PATH = '/var/spool/'
+TMPFS_SIZE = '1M'
+
+class BeaconBroadcasterBroadcastException(Exception):
+    """ This exception is raised when a BeaconBroadcaster object fails to send a beacon."""
+
+class BeaconBroadcasterInitException(Exception):
+    """ This exception is raised when a BeaconBroadcaster object fails to initialize."""
+
+class BeaconBroadcaster(object):
+    """ Provides the broadcasting component of a filesystem-based IPC mechanism."""
+
+    def __init__(self, program_name, beacon_name, uid, gid):
```

Comment:
I think the uid and gid can be determined programmatically.

Reply:
This needs to be initialized as root for directory creation, permissions modification, and tmpfs mounting, but will usually be run as a non-root user. We can't predict the final uid and gid, otherwise I would have done that.

Reply:
Agreed.